tt_metal/impl/dispatch/kernels/cq_common.hpp

```diff
@@ -217,25 +217,23 @@ uint32_t cb_acquire_pages(uint32_t cb_fence,
     volatile tt_l1_ptr uint32_t* sem_addr =
         reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_semaphore(sem_id));
 
-    static uint32_t available = 0;
-
-    if (available == 0) {
-        // Ensure last sem_inc has landed
-        noc_async_atomic_barrier();
+    static uint32_t counter = 0;
+    static uint32_t sem = 0;
```

Comment:
I can rename, though wanted to avoid having upstream as a variable name in this function since it seemed generic/not necessarily upstream. Would remote_count be better?